Fix infinite checkout loader when some script wasn't loaded correctly because of network error

vovayatsyuk opened the pull request:
### Description
Infinite checkout loader may appear when some module makes a `require` call but the dependency wasn't returned (Network error).

The patch changes how Magento treats `isPending` modules. If all dependencies are errored - treat it as loaded.

### Manual testing scenarios
1. Open `Magento/Checkout/view/frontend/web/js/model/quote.js`
2. Insert the following code: 

```js
require([
    'jquery',
    'https://some-site/tracking/api/js'
], function ($) {}, function () {});
``` 

right after

```js
define([
    'ko',
    'underscore'
], function (ko, _) {
    'use strict';

    // add code here
```

3. Navigate to checkout.
4. You can't purchase the product because the page is blocked by loader.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

vovayatsyuk committed:
Do not treat module as pending if dependencies could not be loaded

magento-engcom-team commented:
@vovayatsyuk thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

vovayatsyuk committed:
Proper handling for `require` calls with multiple dependencies:

```
require([
    'ko',
    'https://some-site/tracking/api/js'
], function () {});
```

vovayatsyuk committed:
JSDoc fix

vovayatsyuk committed:
JSCS fix: empty line above ‘return’ is required

magento-engcom-team commented:
Hi @vovayatsyuk. Thank you for your contribution. 
Changes from your Pull Request will be available with the upcoming 2.2.5 release.

eTonki commented:
it still does not load checkout even in version 2.2.5

cbartak-rivet commented:
We are also seeing this issue in 2.2.5, with customers who have ad blockers installed (notably Ad Block Plus).

DrewML commented:
@cbartak-rivet I'd recommend opening a new issue with a stack trace if you can replicate the issue.
